comm1.c

```diff
@@ -294,13 +303,43 @@ write_socket(char *cp, struct object *ob)
 		continue;
 	    }
 
+#ifdef ANSI_COLOR
+	    if (!in_ansi && *cp == ANSI_START)
+            {
+                in_ansi = 1;
+                /*
+                 * Escape won't fall into the range we check below,
+                 * so increment ansi_len now.
+                 */
+                ansi_len++;
+            }
+#endif
 	    len = 1;
 	    while (cp[len] > ' ' && cp[len - 1] != '-')
-		len++;
+            {
+#ifdef ANSI_COLOR
+                if (in_ansi)
+                {
+                    ansi_len++;
 
-	    if (current_column + len >= ip->screen_width)
+                    if (cp[len] == ANSI_END)
+                        in_ansi = 0;
+                }
+#endif
+                len++;
+            }
+
+#ifdef ANSI_COLOR
+            if (current_column + len - ansi_len >= ip->screen_width)
```

Comment:
This is pretty much the pattern. Track how long the sequences are, and subtract that amount when doing length checks.